#########################################################################
# ⭐⭐⭐  Set parameters carefully for satisfied performance!   ⭐⭐⭐ #
#########################################################################

mass        : 0.8 # kg 
gra         : 9.81 
pose_solver : 1     # 0:From ZhepeiWang (drag & less singular) 1:From ZhepeiWang, 2:From rotor-drag    
attitude_fb: 1 # 0: from eth rotors controller; 1: from zju fast-perching controller
ctrl_freq_max   : 100.0
use_bodyrate_ctrl: true

max_manual_vel: 1.0
max_angle: -1  # Attitude angle limit in degree. A negative value means no limit.
low_voltage: 13.2 # 4S battery
full_thrust: 26.8704  # Real net thrust(unit:N) in 100% thrust percentage

pxy_error_max: 0.6
vxy_error_max: 1.0
pz_error_max: 0.3
vz_error_max: 1.0
yaw_error_max: 0.7

use_yaw_rate_ctrl: true
perform_aerodynamics_compensation: false



rc_reverse: # *
    roll: false
    pitch: false
    yaw: false
    throttle: false

auto_takeoff_land:
    enable: true
    enable_auto_arm: true
    no_RC: true
    height: 1.0 # m
    speed: 0.3 # m/s

thrust_model: # The model that maps thrust signal u(0~1) to real thrust force F(Unit:N): F=K1*Voltage^K2*(K3*u^2+(1-K3)*u). 
    print_value: false # display the value of “thr_scale_compensate” or “hover_percentage” during thrust model estimating.
    accurate_thrust_model: false  # This can always enabled if don't require accurate control performance :-)
    # accurate thrust mapping parameters
    K1: 0.7583 # Needs precise calibration!
    K2: 1.6942 # Needs precise calibration!
    K3: 0.6786 # Needs precise calibration! K3 equals THR_MDL_FAC in https://docs.px4.io/master/en/config_mc/pid_tuning_guide_multicopter.html.
    # approximate thrust mapping parameters
    hover_percentage: 0.3  # Thrust percentage in Stabilize/Arco mode # *

hover:
    use_hov_percent_kf: true
    #hzchzc
    percent_lower_limit: 0.15
    percent_higher_limit: 0.87

gain:
    # Cascade PID controller. Recommend to read the code.
    Kp0: 1.6
    Kp1: 1.6
    Kp2: 1.5
    Kv0: 1.6
    Kv1: 1.6
    Kv2: 1.5
    # ↓↓↓ No use now --
    Kvi0: 0.2
    Kvi1: 0.2
    Kvi2: 0.2
    Kvd0: 0.0
    Kvd1: 0.0
    Kvd2: 0.0

    Ka0: 1.0
    Ka1: 1.0
    Ka2: 1.0

    # ↓↓↓ Only used in rate control mode.
    KAngR: 10.0
    KAngP: 10.0
    KAngY: 10.0

track:
    # Cascade PID controller. Recommend to read the code.
    Kp0: 1.2
    Kp1: 1.2
    Kp2: 1.5
    Kv0: 1.6
    Kv1: 1.6
    Kv2: 1.5
    # ↓↓↓ No use now --
    Kvi0: 0.2
    Kvi1: 0.2
    Kvi2: 0.2
    Kvd0: 0.0
    Kvd1: 0.0
    Kvd2: 0.0

    Ka0: 1.0
    Ka1: 1.0
    Ka2: 1.0

    # ↓↓↓ Only used in rate control mode.
    KAngR: 10.0
    KAngP: 10.0
    KAngY: 10.0
    

rotor_drag:  
    x: 0.0  # The reduced acceleration on each axis caused by rotor drag. Unit:(m*s^-2)/(m*s^-1).
    y: 0.0  # Same as above
    z: 0.0  # Same as above
    k_thrust_horz: 0.0 # Set to 0 recommended... --

msg_timeout:
    odom: 0.5
    rc:   0.5
    cmd:  0.5
    imu:  0.5
    bat:  0.5
